cmd/bleve/cmd/zap/docvalue.go

```diff
@@ -255,7 +253,19 @@ func getDocValueLocs(docNum uint64, metaHeader []zap.MetaData) (uint64, uint64)
 		return metaHeader[i].DocNum >= docNum
 	})
 	if i < len(metaHeader) && metaHeader[i].DocNum == docNum {
-		return metaHeader[i].DocDvLoc, metaHeader[i].DocDvLen
+		var start, end uint64
+		if i > 0 {
+			start = metaHeader[i].DocDvOffset
+		}
+		// single element case
+		if i == 0 && len(metaHeader) == 1 {
+			end = metaHeader[i].DocDvOffset
+		} else if i < len(metaHeader)-1 {
+			end = metaHeader[i+1].DocDvOffset
+		} else { // for last element
+			end = start + metaHeader[0].DocDvOffset
```

Comment:
the last element case of using metaHeader[0] feels wrong?

Reply:
if i recall from the meeting last week, this was the solution sreekanth described of having the last offset stored in the first position (which isn't needed because it's implicitly zero.  you had advised against this because of the complexity

Reply:
oh, right -- will leave comment on the other PR #815